src/celeritas/grid/XsCalculator.hh

```diff
@@ -129,24 +128,41 @@ CELER_FUNCTION real_type XsCalculator::operator()(Energy energy) const
     size_type lower_idx = loge_grid_.find(loge);
     CELER_ASSERT(lower_idx + 1 < loge_grid_.size());
 
-    real_type const upper_energy = std::exp(loge_grid_[lower_idx + 1]);
-    real_type upper_xs = this->get(lower_idx + 1);
-    if (lower_idx + 1 == data_.prime_index)
+    real_type lower_energy = std::exp(loge_grid_[lower_idx]);
+    real_type upper_energy = std::exp(loge_grid_[lower_idx + 1]);
+
+    real_type result;
+    if (data_.derivative.empty())
```

Comment:
This class is getting increasingly complicated 😅 In practice do we use all four combinations of {1/E scaling, spline}? I would expect only xs uses the 1/E scaling and only dE/dx and/or range uses spline.

Reply:
I think in practice we'll have xs with all combinations of with/without scaling/spline (only a few have scaling, most have spline) and then range and energy loss without scaling and with spline. What I have here currently isn't correct for the xs with scaling and spline case because it both calculates the derivatives and interpolates using the unscaled values. Maybe for this PR I'll just assert that there's no 1/E scaling instead when calculating the derivatives.

I've been wanting to do some refactoring/renaming of the grids and calculators. I think instead of forcing the scaled xs into a single grid we could use separate grids for the scaled and unscaled values, which would fix the scaling+spline issue as well as the inconsistency in the Rayleigh grid.

Reply:
Yes, indeed that would fix it. I think my decision to force them onto the same grid was premature optimization 😅